app/src/main/scala/org/alephium/explorer/persistence/queries/TransactionQueries.scala

```diff
@@ -368,28 +365,29 @@ trait TransactionQueries extends StrictLogging {
                   gasPrice)
     }
 
-  def getBalanceQuerySQL(address: Address): DBActionSR[(U256, Option[TimeStamp])] = {
+  def getBalanceQueryDEPRECATED(address: Address): DBActionSR[(U256, Option[TimeStamp])] = {
     sql"""
         SELECT outputs.amount, outputs.lock_time
         FROM outputs
-        LEFT JOIN inputs ON outputs.key = inputs.output_ref_key
-        WHERE outputs.main_chain = true AND outputs.address = $address AND inputs.block_hash IS NULL
+        LEFT JOIN inputs
+        ON outputs.key = inputs.output_ref_key
+        WHERE outputs.main_chain = true
+        AND outputs.address = $address
+        AND inputs.block_hash IS NULL
       """.as[(U256, Option[TimeStamp])]
   }
 
-  val getBalanceQuery = Compiled { address: Rep[Address] =>
-    mainOutputs
-      .filter(output => output.address === address)
-      .joinLeft(mainInputs)
-      .on(_.key === _.outputRefKey)
-      .filter(_._2.isEmpty)
-      .map { case (output, _) => (output.amount, output.lockTime) }
-  }
-
-  def getBalanceAction(address: Address): DBActionR[(U256, U256)] = {
-    getBalanceQuery(address).result.map { outputs =>
-      sumBalance(outputs)
-    }
+  def getBalanceQuery(address: Address): DBActionSR[(U256, Option[TimeStamp])] = {
+    sql"""
+      SELECT outputs.amount, outputs.lock_time
```

Comment:
Further optimization: we could sum inside the SQL query, instead of returning the results.

Reply:
I thought about it, but can we sum two different thing? we need 2 sum, everything for the main balance and we also need to sum the one with `lock_time > now`

Reply:
It should be doable in SQL. It will improve the performance when we have 1K+ UTXOs for an address

Reply:
Issue created: https://github.com/alephium/explorer-backend/issues/209

Reply:
One idea is probably to create a function to convert `outputs.lock_time` to `0 or outputs.amount` based on timestamp. Then we sum up both values.